#!/usr/bin/env python

PACKAGE = "singaboat_vrx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gps_offset", double_t, 0, "GPS offset w.r.t. WAM-V along X-axis", 0.85, 0, 2.0)

gen.add("G2G_kP", double_t, 0, "Proportional gain of PID controller", 22.0, 0, 25.0)
gen.add("G2G_kI", double_t, 0, "Integral gain of PID controller", 0.27, 0, 1.0)
gen.add("G2G_kD", double_t, 0, "Derivative gain of PID controller", 2.4, 0, 5.0)
gen.add("G2G_kS", double_t, 0, "Saturation constant of PID controller", 50, 0, 100.0)

gen.add("SK_Vx_kP", double_t, 0, "Proportional gain of PID controller", 1.9, 0, 5.0)
gen.add("SK_Vx_kI", double_t, 0, "Integral gain of PID controller", 0.12, 0, 1.0)
gen.add("SK_Vx_kD", double_t, 0, "Derivative gain of PID controller", 1.8, 0, 5.0)
gen.add("SK_Vx_kS", double_t, 0, "Saturation constant of PID controller", 50, 0, 100.0)

gen.add("SK_Vy_kP", double_t, 0, "Proportional gain of PID controller", 2.2, 0, 5.0)
gen.add("SK_Vy_kI", double_t, 0, "Integral gain of PID controller", 0.12, 0, 1.0)
gen.add("SK_Vy_kD", double_t, 0, "Derivative gain of PID controller", 1.9, 0, 5.0)
gen.add("SK_Vy_kS", double_t, 0, "Saturation constant of PID controller", 50, 0, 100.0)

gen.add("SK_Wz_kP", double_t, 0, "Proportional gain of PID controller", 2.0, 0, 5.0)
gen.add("SK_Wz_kI", double_t, 0, "Integral gain of PID controller", 0.11, 0, 1.0)
gen.add("SK_Wz_kD", double_t, 0, "Derivative gain of PID controller", 1.3, 0, 5.0)
gen.add("SK_Wz_kS", double_t, 0, "Saturation constant of PID controller", 50, 0, 100.0)

gen.add("goal_tol", double_t, 0, "Goal tolerance dead-band of go-to-goal PID controller", 5.0, 0, 10.0)
gen.add("v_const", double_t, 0, "Proportional gain for linear velocity outside goal tolerance", 5.7, 0, 10.0)
gen.add("v_limit", double_t, 0, "Saturation limit for linear velocity outside goal tolerance", 2.23, 0, 2.23)
gen.add("pos_tol", double_t, 0, "Position tolerance to determine whether goal is reached", 0.5, 0, 1.0)
gen.add("rot_tol", double_t, 0, "Orientation tolerance to determine whether goal is reached", 0.1, 0, 0.5)

gen.add("debug", bool_t, 0, "Flag to enable/disable debug messages", True)

exit(gen.generate(PACKAGE, "singaboat_vrx", "Wayfinding"))
